### ISSUE TITLE
machines: Change naming conventions of Networks and Disks

### ISSUE THREAD

**Issue by skobyda** [cockpit-project/cockpit#11171]
Page: machines

Right now machines contain "Networks" tab for each guest which deals with virtual interfaces, then there is "Networks" Status card (with it's own page in development) which deals with virtual networks, and also there is separate plugin "Networking" which deals with physical interfaces and networks.

It is a bit confusing they all bear same name.
So I suggest to rename "Networks" tab to "Interfaces" tab.

Also if we want to give components accurate names, we could also rename "Disks" -> "Storage", and "Networks" (the Status Card) -> "Virtual networks".

**Comment by skobyda** [comment]
@garrett @andreasn @KKoukiou What do you think?

**Comment by andreasn** [comment]
Thanks for raising this.
I'm still trying to wrap my head around the relationship between the host network interfaces handled by networkmanager the virtual network and the VMs guest network and how they fit together, so I'll need to read up a little bit more on that. I would have no idea where to even start if I wanted expose a VM to my wider network, or have it talk to another VM.

For Networks. Is your worry that the card will lead the administrator using the system to think that card will take them to the hosts Networking page instead? If so, maybe Virtual could be added to the string, but at the same time, isn't everything virtual on the VM page? Virtual Memory, Virtual CPU, Virtual Storage etc? It might end up being a bit redundant.

So with regards to Disks vs Storage. On the Virtual Storage list, I have Pools and inside the pools there are Storage Volumes. These seems to match up to the items under the VMs Disks tab. Does that mean that the tab under the VM should be called Volumes instead? I think we'll need to sit down and sort out all the storage related vocabulary across the VMs page, because it slipped apart a bit over time.

**Comment by KKoukiou** [comment]
Quoting libvirt docs for better understanding:

`
A storage pool is a quantity of storage set aside by an administrator, often a dedicated storage administrator, for use by virtual machines. Storage pools are divided into storage volumes either by the storage administrator or the system administrator, and the volumes are assigned to VMs as block devices. (https://libvirt.org/storage.html)
`

However storage volume is not equivalent of a disk. A volume is not a disk untill it's attached to some guest.
And vise versa not all disks are storage volumes. A disk can have as a source a file which is not  a storage volume for libvirt. And this is just because noone defined that file as a volume.

The format for disk XML is defined here https://libvirt.org/formatdomain.html#elementsDisks
You can see there that the volume element and pool element are not mandatory.

**Comment by skobyda** [comment]
So basically in libvirt virtual interface works like a physical interface for guest. Through virtual interface, guest can connect to virtual network or directly to physical network (e.g. bridge).

However virtual network is different thing. Virtual network can be connected to your physical interface from one side. And many virtual guests can connect their virtual interfaces to the virtual network from the other side. 
Point of all that is when your guest wants to connect to internet for example, your host only needs one connection to the outside physical network, and then the virtual network takes care that correct packets from internet are send to correct virtual interface of correct virtual guest.
Even through all the packets from physical network come through one physical interface, it reaches the virtual interface of correct guest, because of virtual network.

In summary:
Interface is like frontend and network is like a backend for connection. :)

And that's why I think "networks" tab should be renamed to "interfaces" tab.

> isn't everything virtual on the VM page
Right. I'm not so keen on renaming "Networks" to "Virtual networks"

> However storage volume is not equivalent of a disk.
Well yeah, but we do not want to rename "Disks" to "Storage volume". Only to "Storage". But it is right that every devices that VM uses for **storage** is a **disk** (whetever it's storage volume, CD, floppy...). So I guess both "Disks" or "Storage" are correct names for that tab.

**Comment by skobyda** [comment]
So since changing Networks to Interfaces didn't have any opposition, I would send Pull Request for it. In the UI, it would reflext in three places:
Name of the "Networks" tab would change to "Interfaces" tab
![screenshot from 2019-02-20 17-12-36](https://user-images.githubusercontent.com/42733240/53109475-3fb8e280-3539-11e9-862a-287edd4c80a3.png)
In the Interface Edit Modal, "Network Type" to "Interface Type", and in Error Message "Network settings could not be saved" to "Interface settings could not be saved".
![screenshot from 2019-02-20 17-12-59](https://user-images.githubusercontent.com/42733240/53109836-08970100-353a-11e9-8352-f8980074e399.png)

**Comment by KKoukiou** [comment]
@Antique i would like that you approve this change before happening, you have better insight for this than me.

**Comment by Antique** [comment]
`Interface` is probably a better name than `Network`, however, technically correct name would be `Network interface` as interface is just a generic term.

**Comment by andreasn** [comment]
Yeah, _interface_ can mean a lot of things. Like a _user interface_, unless it's context hints at what kind of interface it is (as it does on the host networking page). I think _Network Interface_ sounds suitable here.

**Comment by skobyda** [comment]
Network Interface it is! :)
